src/main/java/com/appoptics/integrations/kafka/broker/KafkaAppopticsReporter.java

```diff
@@ -17,18 +20,46 @@
     private Reporter.Builder reporterBuilder;
     private Reporter reporter;
 
+    private final String URL = "appoptics.url";
+    private final String TOKEN = "appoptics.token";
+    private final String AGENT_IDENTIFIER = "appoptics.agent.identifier";
+    private final String TAGS = "appoptics.tags";
+
     @Override
     public synchronized void init(VerifiableProperties props) {
-        String apiUrl = props.getString("appoptics.url");
+        String apiUrl = props.getString(URL);
         if (apiUrl == null) {
             apiUrl = "https://api.appoptics.com/v1/measurements";
         }
 
-        String token = props.getString("appoptics.token");
-        String source = props.getString("appoptics.agent.identifier");
+        String token = props.getString(TOKEN);
+        List<Tag> tags = new ArrayList<Tag>();
+
+        String source = props.getString(AGENT_IDENTIFIER, "");
+        if (!source.isEmpty()) {
+            tags.add(new Tag("source", source));
+        }
+
+        String customTags = props.getString(TAGS, "");
+        if (!customTags.isEmpty()) {
+            String[] str = new String[]{customTags};
+            if (customTags.contains(",")) {
+                str = customTags.split(",");
+            }
+            for (String tagString : str) {
+                if (customTags.contains(":")) {
+                    String properties[] = tagString.split(":");
+                    if (str.length == 2) {
+                        Tag tag = new Tag(properties[0], properties[1]);
```

Comment:
We might also want to consider testing the validity of tag name and value before adding these as an invalid name or value will result in the measurement being rejected.

Reply:
Hi @petercable can you like me to some documentation which would describe what could make the tag name invalid?

Reply:
@Beaujr it's just a bit further down the page here: https://docs.appoptics.com/api/#measurement-restrictions

tag names: `/\A[-.:_\w]{1,64}\z/` (up to 64 characters, a-z, A-Z, 0-9, -, ., :, and _)
tag values: `/\A[-.:_?\\\/\w ]{1,255}\z/` (up to 255 characters, a-z, A-Z, 0-9, -, ., :, _, ?, \, and /)